### PR TITLE
fix(VDataIterator): add isMobile property to custom item slot scope

### LINKED ISSUE #9436: [Bug Report] data-table mobile view with custom slots
### Environment
**Vuetify Version:** 2.1.5
**Vue Version:** 2.6.10
**Browsers:** Chrome 76.0.3809.87
**OS:** Linux x86_64

### Steps to reproduce
add custom item slot to data-table

### Expected Behavior
the two table should render the same in mobile view

### Actual Behavior
the table with custom slot column doesn't stack

### Reproduction Link
<a href="https://codepen.io/AshrafBasry/pen/QWWdeML" target="_blank">https://codepen.io/AshrafBasry/pen/QWWdeML</a>

### Other comments
i searched in the data-tables docs for how to make this work but i couldn't find any thing about mobile view.

and why slot items props.index is no longer refer to items index like the previous versions and refer to the visible rows index !!

### PR DESCRIPTION
**Pull request by alicjamusial**
Add possibility to customize 'item' slot in DataTable and DataIterator in mobile view by introducing
isMobile property to be available in the scope.

fix #9436

## Description

Custom `item` slots in DataTable render badly on mobiles. Header in table is responsive but the slot is not, which causes it to look really bad (https://codepen.io/AshrafBasry/pen/QWWdeML). The proposed solution is the easiest possible - by adding `isMobile` prop into the scope, we can allow the developer to customize their `item` slot both for normal and mobile view.

## Motivation and Context
https://github.com/vuetifyjs/vuetify/issues/9436
https://github.com/vuetifyjs/vuetify/issues/8394

## How Has This Been Tested?
Visually.

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-data-table :headers="tableHeaders" :items="tableData">
      <template v-slot:item="{item, isMobile}">
        <tr v-if="!isMobile">
          <td>{{item.col1}}</td>
          <td>{{item.col2}}</td>
          <td>{{item.col3}}</td>
          <td>{{item.col4}}</td>
        </tr>
        <tr v-else>
          <td>Custom mobile view - {{item.col1}}</td>
        </tr>
      </template>
    </v-data-table>

    <v-data-iterator :items="tableData">
      <template v-slot:item="{item, isMobile}">
        <v-card v-if="!isMobile" class="my-2">
          <v-card-title class="subheading font-weight-bold">
            {{item.col1}}
          </v-card-title>
          <v-card-actions>
            <p>Desktop item view</p>
          </v-card-actions>
        </v-card>
        <v-card v-else class="my-2">
          <v-card-title class="subheading font-weight-bold">
            {{item.col1}}
          </v-card-title>
          <v-card-actions>
            <p>Mobile item view</p>
          </v-card-actions>
        </v-card>
      </template>
    </v-data-iterator>
  </v-container>
</template>

<script>
export default {
  data() {
    return {
      tableHeaders: [
          {
            text: 'Col 1',
            sortable: true,
            value: 'col1'
          }, {
            text: 'Col 2',
            sortable: false,
            value: 'col2'
          },{
            text: 'Col 3',
            sortable: false,
            value: 'col3'
          }, {
            text: 'Col 4',
            sortable: false,
            value: 'col4'
          }
        ],
      tableData: [
          {
            col1: 'Data 1',
            col2: "Data 2",
            col3: "Data 3",
            col4: "Data 4",
          },
          {
            col1: 'Data 1',
            col2: "Data 2",
            col3: "Data 3",
            col4: "Data 4",
          },
          {
            col1: 'Data 1',
            col2: "Data 2",
            col3: "Data 3",
            col4: "Data 4",
          },
        ]
    }
  }
}
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VDataIterator/__tests__/VDataIterator.spec.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import {
 } from '@vue/test-utils'
 import Vue from 'vue'
 import { preset } from '../../../presets/default'
+import {Breakpoint} from "../../../services/breakpoint";
```
Comment (nekosaur) [inline packages/vuetify/src/components/VDataIterator/__tests__/VDataIterator.spec.ts]: space before and after Breakpoint

**Comment by alicjamusial** [comment]
> 
> 
> VDataTable extends VDataIterator, so if you're adding mobileBreakpoint and isMobile to iterator, remove them from VDataTable (note that you're missing a check in isMobile compared to data-table).

Done! Sorry for the multiple commits - my first PR :). If there is anything more I should change/refactor, please let me know.

File: packages/vuetify/src/components/VDataIterator/VDataIterator.ts [hunk 2]
```diff
@@ -32,6 +32,10 @@ export default Themeable.extend({
       type: Array as PropType<any[]>,
       default: () => [],
     },
+    mobileBreakpoint: {
+      type: Number,
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataIterator/VDataIterator.ts]:
```suggestion
      type: [Number, String],
```

File: packages/vuetify/src/components/VDataIterator/VDataIterator.ts [hunk 3]
```diff
@@ -73,6 +77,13 @@ export default Themeable.extend({
     selectableItems (): any[] {
       return this.internalCurrentItems.filter(item => this.isSelectable(item))
     },
+    isMobile (): boolean {
+      // Guard against SSR render
+      // https://github.com/vuetifyjs/vuetify/issues/7410
+      if (this.$vuetify.breakpoint.width === 0) return false
+
+      return this.$vuetify.breakpoint.width < this.mobileBreakpoint
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataIterator/VDataIterator.ts]:
```suggestion
      return this.$vuetify.breakpoint.width < parseInt(this.mobileBreakPoint, 10)
```

File: packages/vuetify/src/components/VDataIterator/__tests__/VDataIterator.spec.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import {
 } from '@vue/test-utils'
 import Vue from 'vue'
 import { preset } from '../../../presets/default'
+import { Breakpoint } from '../../../services/breakpoint'
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataIterator/__tests__/VDataIterator.spec.ts]: should be above line 9

**Comment by johnleider** [comment]
@nekosaur just need your final approval

**Pull request merged**